repair/row_level.cc

```diff
@@ -2934,6 +2965,7 @@ repair_service::repair_service(distributed<gms::gossiper>& gossiper,
     , _node_ops_metrics(_repair_module)
     , _max_repair_memory(max_repair_memory)
     , _memory_sem(max_repair_memory)
+    , _reader_sem(10)
```

Comment:
What is this magic number 10 - is it the number of readers on each shard? Something else? And how did you choose it?

Reply:
It is the magic 10 reader limit allowed for readers for streaming. It is defined:

static constexpr size_t max_count_streaming_concurrent_reads{10};

It is not exposed currently. I plan to expose in a follow up. Another on possible future work is to increase this limit. 

So I am not touching it in this PR.

Reply:
If it's supposed to be identical to another variable which is not visible here and you want to fix it in the future, can you please add a FIXME comment, e.g.,
```
// The "10" below should be the same mas max_count_streaming_concurrent_reads. FIXME: use that named constant instead of the number here.
```

Reply:
Done.

Reply:
You can also just use `_db.local().get_reader_concurrency_semaphore().initial_resources().count`, because the count is the same on all shards. Note that `get_reader_concurrency_semaphore()` returns the scheduling-group approriate semaphore, which will be the streaming one here (you can verify with debug log, printing `name()` of semaphore and the count above to verify it is indeed 10).

Reply:
The code to initialize repair_service could be run outside the maintenance group. This could give wrong semaphore. I would avoid depending on the current_scheduling_group() if possible, considering the scheduling leak bug we had in the past  which was super hard to figure out. 

```
request_class classify_request(const database_config& _dbcfg) {
    const auto current_group = current_scheduling_group();

    // Everything running in the statement group is considered a user request
    if (current_group == _dbcfg.statement_scheduling_group) {
        return request_class::user;
    // System requests run in the default (main) scheduling group
    // All requests executed on behalf of internal work also uses the system semaphore
    } else if (current_group == default_scheduling_group()
            || current_group == _dbcfg.compaction_scheduling_group
            || current_group == _dbcfg.gossip_scheduling_group
            || current_group == _dbcfg.memory_compaction_scheduling_group
            || current_group == _dbcfg.memtable_scheduling_group
            || current_group == _dbcfg.memtable_to_cache_scheduling_group) {
        return request_class::system;
    // Requests done on behalf of view update generation run in the streaming group
    } else if (current_scheduling_group() == _dbcfg.streaming_scheduling_group) {
        return request_class::maintenance;
    // Everything else is considered a user request
    } else {
        return request_class::user;
    }
}
```

Reply:
Ok, you can access the streaming semaphore directly then. If there is no getter for it, you can add one.

Reply:
For the record, I played with
```
--- a/repair/row_level.cc
+++ b/repair/row_level.cc
@@ -2977,6 +2977,7 @@ repair_service::repair_service(distributed<gms::gossiper>& gossiper,
     // FIXME: use that named constant instead of the number here.
     , _reader_sem(10)
 {
+    rlogger.info("repair_service: sem name = {}", _db.local().get_reader_concurrency_semaphore().name());
```

It indeed gave wrong semaphore.

```
INFO  2022-12-05 16:44:54,385 [shard 0] init - starting repair service
INFO  2022-12-05 16:44:54,385 [shard 0] repair - Setting max_repair_memory=107374182, max_repair_memory_per_range=33554432, max_repair_ranges_in_parallel=1
INFO  2022-12-05 16:44:54,385 [shard 0] repair - repair_service: sem name = _system_read_concurrency_sem
```